refinery/file_store/tasks.py

```diff
@@ -42,39 +39,101 @@ def import_file(uuid, refresh=False, file_size=0):
             FileStoreItem.MultipleObjectsReturned) as exc:
         logger.error("Error importing FileStoreItem with UUID '%s': %s",
                      uuid, exc)
-        return None
+        import_file.update_state(state=celery.states.FAILURE,
+                                 meta='Failed to import file')
+        # http://stackoverflow.com/a/33143545
+        raise celery.exceptions.Ignore()
+
+    # exit if an import task is already running for this file
+    if item.import_task_id:
+        result = celery.result.AsyncResult(item.import_task_id)
+        if result.state in list(celery.states.UNREADY_STATES) + ['PROGRESS']:
+            logger.error(
+                "File import is already in progress for '%s' - task ID: '%s'",
+                item, item.import_task_id
+            )
+            import_file.update_state(state=celery.states.FAILURE,
```

Comment:
It seems like this pattern:
```
import_file.update_state(state=celery.states.FAILURE,
                         meta='Failed to import file')
# http://stackoverflow.com/a/33143545
raise celery.exceptions.Ignore()
```
is repeated in a few places within `import_file`. Would it be worth factoring out?

Reply:
The whole `import_file` needs to be split up but this is a [separate issue](https://github.com/refinery-platform/refinery-platform/issues/2359).

Reply:
I think having these ~10 similar chunks of code is just creating more overhead for when https://github.com/refinery-platform/refinery-platform/issues/2359 is tackled, but its ultimately up to you.